### PR TITLE
SparkleTilt PCB workshop

### PR DESCRIPTION
**Pull request by karmanyaahm**
For these workshops, I'm making a Level that lights LEDs based on a the angle. It can also serve as a decoration. To make it less intimidating, I'm breaking it down into parts doing different things. People should also be able to take ideas from individual parts to use in their projects (just the microcontroller, just the BMC, etc).

Plan:
1. Schematic for a microcontroller board - **Ready for review**: <https://workshops.hackclub.com/external-preview/karmanyaahm/onboard/pcb_level_1/>
2. PCB Design (KiCAD) - **Ready for review**: <https://workshops.hackclub.com/external-preview/karmanyaahm/onboard/pcb_level_2/>
4. Ordering from JLCPCB - **Ready for review**: <https://workshops.hackclub.com/external-preview/karmanyaahm/onboard/pcb_level_3/>
3. PCB Design (EasyEDA) - **No for now ig**
6. Peripherals (Accelerometer, LEDs, Battery Stuff) - **Ready for review** - <https://workshops.hackclub.com/external-preview/karmanyaahm/onboard/pcb_level_4/>

TODO: 
- [ ] [Footnotes are not linked](https://github.com/hackclub/markdown/issues/15) 
- [x] CDN-ify images
- [x] Better workshop titles?

### CODE REVIEW

File: workshops/pcb_level_1/README.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+</details>
+
+
+
+I am using: 
+<label><input name="viewSettings" type="radio" data-to-hide=".kicad-img" checked=true/> KiCAD</label>
+<label><input name="viewSettings" type="radio" data-to-hide=".easyeda-img" />EasyEDA</label>
+<script>
+    // this script will switch between KiCAD images and EasyEDA images based on radio buttons above
+
```
Comment (karmanyaahm) [inline workshops/pcb_level_1/README.md:29]: This little bit of JS renders as:
image
and allows the reader to select between images for KiCAD and EasyEDA. Is this fine?

Comment (karmanyaahm) [reply workshops/pcb_level_1/README.md:29]: Looks like this: https://workshops.hackclub.com/external-preview/karmanyaahm/onboard/pcb_level_1/

**Comment by Hugoyhu** [comment]
A few comments I had as I went through the KiCAD version for Part 1

- ATmega328P is available in a variety of form factors- DIP28, TQFP, and QFN- very flexible. may want to mention that you need the ATmega328P not _PB_ variant. They're not backwards compatible. 
- Decoupling capacitors also absorb voltage spikes. 
- Crystal Y2 is one of the "4 pin variants" and not the 2 pin ones. This might pose a problem based on which one you're using. Because some do GND on 2, 4 some do GND on 1, 3. Add. some resonators use exact same package, adding complexity if you don't know diff. 
- People usually put R6 as 10K, not 100K. 
- CH340N schematic is very messy- probably easier to read if you used labels and put it separately somewhere

**Comment by Hugoyhu** [comment]
Comments for Part 2:
- I've never actually set the DRC specs, the default ones are pretty much universal for any competent PCB fab. This is a lot of steps that might get confusing.
- You can also use the Bitmap2Component tool in case you don't have a nice SVG.
- Your SVG might not always be the right size. This might take some trial and error so maybe mention that. Usually when doing PCB art, I do a rough (unrouted) layout and then calculate the sizing needed with measure tool
- Typo for "rotate them to a position... **convinient**"
- Maybe do the ground plane before starting to route.
- Your routing design really destroys the ground plane essentially where it's really needed, so having the ground plane there can help influence better routing decisions (keeping everything mostly one layer, don't disrupt it on the other layer with less routes)
- Make a note that "errors" may not actually be errors. This might make it really intimidating to newbies. Usually they're just like "blah blah blah silkscreen overlap" like idc lol 
- 1) unconnected 2) major errors  etc
- For sidenote 2: the unique layout that's forced here kinda does that to you. placing the parts around the board more spread out would alleviate this issue.

**Comment by Hugoyhu** [comment]
Part 3 comments:
- you should try to write this as "generate a Gerber", etc. Not everyone has a plugin, and this kind of limits your purchasing. Some US/EU based fabs like Oshpark, Aisler have nice things that accept KiCAD (and maybe other formats like eagle, etc). But Chinese ones don't care much, and gerbers remain the industry standard, no matter where you are. 
- Similar to this, exporting BOM and pos files are a much better practice in case you can't order with JLC. 
- Soldering a USBC connector by hand isn't exactly nice unless you extend the pads, which isn't done here
- You don't actually put any pins for A1 and J2, meaning that unless if you have your soldering setup + female and male headers, you can't really use the board

**Comment by karmanyaahm** [comment]
(working on editing with your comments Hugo)
As I add peripherals I'm considering 4-layering the board. It's becoming a nightmare to route the different power lines for the components, charger/LEDs, and I2C lines, etc, etc.

**Comment by karmanyaahm** [comment]
Actually nvm 4-layer is $20 more

**Comment by karmanyaahm** [comment]
> A few comments I had as I went through the KiCAD version for Part 1

3 - Unfortunately JLCPCB only has this in their 16MHz basic parts. I'll clarify selecting the right version in part selection.

Fixed all the other points.

**Comment by karmanyaahm** [comment]
> Comments for Part 2:
> 
>     * I've never actually set the DRC specs, the default ones are pretty much universal for any competent PCB fab. This is a lot of steps that might get confusing.

That's a good point, I didn't notice earlier that clearance and stuff was set by Net Classes already. Do you think I should still keep the "0.3mm for signal traces, 0.5mm for power" thing? Or should I leave them at the default, .25mm.

>     * Maybe do the ground plane before starting to route.
> 
>     * Your routing design really destroys the ground plane essentially where it's really needed, so having the ground plane there can help influence better routing decisions (keeping everything mostly one layer, don't disrupt it on the other layer with less routes)

I did keep it in mind when routing, but short of making the signals jump back and forth with vias even more, there wasn't much I could do. I will mention minimizing the use of the Back layer though.

**Comment by karmanyaahm** [comment]
> Part 3 comments:
> 
>     * you should try to write this as "generate a Gerber", etc. Not everyone has a plugin, and this kind of limits your purchasing. Some US/EU based fabs like Oshpark, Aisler have nice things that accept KiCAD (and maybe other formats like eagle, etc). But Chinese ones don't care much, and gerbers remain the industry standard, no matter where you are.
> 
>     * Similar to this, exporting BOM and pos files are a much better practice in case you can't order with JLC.

I agree that another generic guide would be helpful, but JLCPCB has weird [requirements](https://jlcpcb.com/help/article/81-How-to-generate-the-BOM-and-Centroid-file-from-KiCAD) for BOM and Position files. The plugin makes that stuff easier. 
This guide also shows adding LCSC part numbers. And, since JLCPCB is the main manufacturer for OnBoard that offers assembly, most people following this guide will use them anyway. So having a JLCPCB-specific guide would be the most straightforward. A separate guide for other manufacturers would be very helpful.

>     * You don't actually put any pins for A1 and J2, meaning that unless if you have your soldering setup + female and male headers, you can't really use the board

Good point. I originally intended this to be a starting point for more complex PCBs but I can see it being used like that. I'll add a note with the process for headers.

**Comment by karmanyaahm** [comment]
moving to https://github.com/hackclub/jams/pull/176

**Pull request closed without merging**